bentoml/_internal/frameworks/pickle.py

```diff
@@ -0,0 +1,285 @@
+        runner.run(3)
+
+    """  # noqa
+    context = {"framework_name": "pickle"}
+
+    _model = Model.create(
+        name,
+        module=MODULE_NAME,
+        metadata=metadata,
+        context=context,
+    )
+
+    with open(_model.path_of(f"{SAVE_NAMESPACE}{PKL_EXT}"), "wb") as f:
+        cloudpickle.dump(model, f)
+
+    _model.save(model_store)
+    return _model.tag
+
+
+class _PickleRunner(Runner):
+    @inject
+    def __init__(
+        self,
+        tag: Tag,
+        function_name: str,
+        name: str,
+        resource_quota: t.Optional[t.Dict[str, t.Any]],
+        batch_options: t.Optional[t.Dict[str, t.Any]],
+        model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
+    ):
+        super().__init__(name, resource_quota, batch_options)
+        model_info, model_file = _get_model_info(tag, model_store)
+        self._model_store = model_store
+        self._model_info = model_info
+        self._model_file = model_file
+        self._backend = "loky"
+        self._function_name = function_name
+
+    @property
+    def num_replica(self) -> int:
+        return 1
+
+    @property
+    def required_models(self) -> t.List[Tag]:
+        return [self._model_info.tag]
+
+    # pylint: disable=attribute-defined-outside-init
+    def _setup(self) -> None:
+        with open(self._model_file, "rb") as f:
+            self._model = pickle.load(f)
+        self._infer_func = getattr(self._model, self._function_name)
+
+    # pylint: disable=arguments-differ
+    def _run_batch(  # type: ignore[reportIncompatibleMethodOverride]
+        self,
+        inputs: t.Union["ext.NpNDArray", "ext.PdDataFrame"],
+    ) -> "ext.NpNDArray":
+        return self._infer_func(inputs)
+
+
+class _PickleSimpleRunner(SimpleRunner):
+    @inject
+    def __init__(
+        self,
+        tag: Tag,
+        function_name: str,
+        name: str,
+        resource_quota: t.Optional[t.Dict[str, t.Any]],
+        batch_options: t.Optional[t.Dict[str, t.Any]],
+        model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
+    ):
+        super().__init__(name, resource_quota, batch_options)
+        model_info, model_file = _get_model_info(tag, model_store)
+        self._model_store = model_store
+        self._model_info = model_info
+        self._model_file = model_file
+        self._backend = "loky"
+        self._function_name = function_name
+
+    @property
+    def num_replica(self) -> int:
+        return 1
+
+    @property
+    def required_models(self) -> t.List[Tag]:
+        return [self._model_info.tag]
+
+    # pylint: disable=attribute-defined-outside-init
+    def _setup(self) -> None:
+        with open(self._model_file, "rb") as f:
+            self._model = pickle.load(f)
+        self._infer_func = getattr(self._model, self._function_name)
+
+    def _run(  # type: ignore[reportIncompatibleMethodOverride]
+        self,
+        inputs: t.Union["ext.NpNDArray", "ext.PdDataFrame"],
```

Comment:
We might want to just put `t.Any` in and `t.Any` out? Not sure how this is really meant to be used.

Reply:
yes this should be just `t.Any` since SimpleRunner does not need the batching logic in DataContainer and support any parameter that can be pickled (for senting between api server and runner processes).

Reply:
done